### PR TITLE
Task 2

### PR DESCRIPTION
**Pull request by artyom123**

### CODE REVIEW

File: task2/src/services/users.ts [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        });
+    } else {
+        res.status(200).json(user);
+    }
+};
+
+export const findAll = (req: Request, res: Response) => {
+    const { limit = users.length, loginSubstring = '' }: any = req.query;
+
+    const result: User[] = users.filter((user, index) => user.login.includes(loginSubstring)  && index < limit)
```
Comment (potsiahevich-dmitry) [inline task2/src/services/users.ts:24]: I believe that we should get all the filtered results and only after sorting them cut the requested limit

File: task2/src/services/users.ts [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+
+    if (!user) {
+        res.status(404).json({
+            error: 'User not exists'
+        });
+    } else {
+        const { password, age } = req.body;
+
+        Object.assign(
```
Comment (potsiahevich-dmitry) [inline task2/src/services/users.ts:52]: Assigning new properties to the array of users?

File: task2/src/services/users.ts [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+        );
+
+        res.status(200).json(users);
+    }
+};
+
+export const create = (req: Request, res: Response) => {
+    const { login, password, age } = req.body;
+
+    const isLogin = users.find((userItem) => userItem.login === login);
```
Comment (potsiahevich-dmitry) [inline task2/src/services/users.ts:68]: isExist/isCreated seems to be a better name instead of isLogin

**Review by potsiahevich-dmitry — CHANGES_REQUESTED** [review]

File: task2/src/services/users.ts (hunk 1 above)
Comment (artyom123) [reply task2/src/services/users.ts]: Ok, I fixed this.

File: task2/src/services/users.ts (hunk 2 above)
Comment (artyom123) [reply task2/src/services/users.ts]: Fixed

File: task2/src/services/users.ts (hunk 3 above)
Comment (artyom123) [reply task2/src/services/users.ts]: Fixed

**Review by potsiahevich-dmitry — APPROVED** [review]

**Pull request merged**